### PR TITLE
fix(trigger_child_pipeline): Add a new `timeout` argument to fail fast in case of issue

### PR DESCRIPTION
**Pull request by chouetz**
### What does this PR do?
Add a new `timeout` variable to be able to finish jobs earlier if jobs are stuck (fail fast)

### Motivation
Initially the pipeline timeout when using the CLI was 5h. As we have a maximum timeout of 2h per gitlab job definition the `trigger-child-pipeline` timeout should be 2h at maximum.
Looking at the maximum duration of the following pipeline
- [agent-release-management](https://app.datadoghq.com/ci/pipeline-executions?query=ci_level%3Apipeline%20env%3Aprod%20%40git.repository.id%3Agitlab.ddbuild.io%2FDataDog%2Fagent-release-management%20%40ci.pipeline.name%3ADataDog%2Fagent-release-management%20%40ci.provider.instance%3Agitlab-ci%20%40git.branch%3Amain%20%40ci.pipeline.fingerprint%3AtBuFoSLT9_ks%20%40gitlab.upstream.project%3A%22DataDog%2Fagent-release-management%22&agg_m=%40duration&agg_m_source=base&agg_t=avg&cipipeline_explorer_sort=%40duration%2Cdesc&cols=%40ci.status%2Ctimestamp%2C%40ci.pipeline.name%2C%40duration%2C%40ci.pipeline.id%2C%40git.repository.name%2C%40git.branch%2C%40ci_stages&fromUser=false&index=cipipeline&viz=stream&start=1717242365913&end=1719834365913&paused=false)
- [agent-linux-install-script](https://app.datadoghq.com/ci/pipeline-executions?query=ci_level%3Apipeline%20env%3Aprod%20%40git.repository.id%3Agitlab.ddbuild.io%2FDataDog%2Fagent-linux-install-script%20%40ci.pipeline.name%3ADataDog%2Fagent-linux-install-script%20%40ci.provider.instance%3Agitlab-ci%20%40git.branch%3Amain%20%40gitlab.upstream.project%3A%22DataDog%2Fdatadog-agent%22&agg_m=%40duration&agg_m_source=base&agg_t=avg&cipipeline_explorer_sort=%40duration%2Cdesc&cols=%40ci.status%2Ctimestamp%2C%40ci.pipeline.name%2C%40duration%2C%40ci.pipeline.id%2C%40git.repository.name%2C%40git.branch%2C%40ci_stages&fromUser=false&index=cipipeline&viz=stream&start=1717242539245&end=1719834539245&paused=false)
- [Images](https://app.datadoghq.com/ci/pipeline-executions?query=ci_level%3Apipeline%20env%3Aprod%20%40git.repository.id%3Agitlab.ddbuild.io%2FDataDog%2Fimages%20%40ci.pipeline.name%3ADataDog%2Fimages%20%40ci.provider.instance%3Agitlab-ci%20%40git.branch%3Amaster%20%40gitlab.upstream.project%3A%22DataDog%2Fdatadog-agent%22&agg_m=%40duration&agg_m_source=base&agg_t=avg&cipipeline_explorer_sort=%40duration%2Cdesc&cols=%40ci.status%2Ctimestamp%2C%40ci.pipeline.name%2C%40duration%2C%40ci.pipeline.id%2C%40git.repository.name%2C%40git.branch%2C%40ci_stages&fromUser=false&index=cipipeline&viz=stream&start=1717243019391&end=1719835019391&paused=false)
- [public-images](https://app.datadoghq.com/ci/pipeline-executions?query=ci_level%3Apipeline%20env%3Aprod%20%40git.repository.id%3Agitlab.ddbuild.io%2FDataDog%2Fpublic-images%20%40ci.pipeline.name%3ADataDog%2Fpublic-images%20%40ci.provider.instance%3Agitlab-ci%20%40git.branch%3Amain%20%40gitlab.upstream.project%3A%22DataDog%2Fdatadog-agent%22&agg_m=%40duration&agg_m_source=base&agg_t=avg&cipipeline_explorer_sort=%40duration%2Cdesc&cols=%40ci.status%2Ctimestamp%2C%40ci.pipeline.name%2C%40duration%2C%40ci.pipeline.id%2C%40git.repository.name%2C%40git.branch%2C%40ci_stages&fromUser=false&index=cipipeline&viz=stream&start=1717243126421&end=1719835126421&paused=false)
- [k8s-datadog-agent-ops](https://app.datadoghq.com/ci/pipeline-executions?query=ci_level%3Apipeline%20env%3Aprod%20%40git.repository.id%3Agitlab.ddbuild.io%2FDataDog%2Fk8s-datadog-agent-ops%20%40ci.pipeline.name%3ADataDog%2Fk8s-datadog-agent-ops%20%40ci.provider.instance%3Agitlab-ci%20%40git.branch%3Amain%20%40gitlab.upstream.project%3A%22DataDog%2Fk8s-datadog-agent-ops%22&agg_m=%40duration&agg_m_source=base&agg_t=avg&cipipeline_explorer_sort=%40duration%2Cdesc&cols=%40ci.status%2Ctimestamp%2C%40ci.pipeline.name%2C%40duration%2C%40ci.pipeline.id%2C%40git.repository.name%2C%40git.branch%2C%40ci_stages&fromUser=false&index=cipipeline&viz=stream&start=1717243679278&end=1719835679278&paused=false) 

We call the invoke task with more restricted timeouts when possible

### CODE REVIEW

File: tasks/pipeline.py [hunk 1]
```diff
@@ -462,7 +462,7 @@ def trigger_child_pipeline(_, git_ref, project_name, variable=None, follow=True)
     if follow:
         print("Waiting for child pipeline to finish...", flush=True)
 
-        wait_for_pipeline(repo, pipeline, pipeline_finish_timeout_sec=2700)
+        wait_for_pipeline(repo, pipeline, pipeline_finish_timeout_sec=7200)
```
Comment (pducolin) [inline tasks/pipeline.py:465]: **💬 suggestion**
```suggestion
		# child timeout is 2h = 2*60*60 = 7200sec
        wait_for_pipeline(repo, pipeline, pipeline_finish_timeout_sec=7200)
```

Comment (chouetz) [reply tasks/pipeline.py:465]: Taking into account another comment, let's add various timeouts based on the pipeline we trigger

File: tasks/pipeline.py [hunk 2]
```diff
@@ -401,7 +401,7 @@ def wait_for_pipeline_from_ref(repo: Project, ref):
 
 
 @task(iterable=['variable'])
-def trigger_child_pipeline(_, git_ref, project_name, variable=None, follow=True):
+def trigger_child_pipeline(_, git_ref, project_name, variable=None, follow=True, timeout=7200):
```
Comment (chouquette) [inline tasks/pipeline.py:404]: Is 7200 the expected value? The previous default was 2700 but I'm not sure which one is a typo for the other :sweat_smile:

Comment (chouetz) [reply tasks/pipeline.py:404]: yes 7200 (=2h) is the default as equal to the job timeout duration. 2700 was 45min, and as you mentioned yesterday it was too short

**Review by chouquette — APPROVED** [review]

**Review by CelianR — APPROVED** [review]
👏

**Pull request merged**